platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/creation/SyncEventCreationManager.java

```diff
@@ -64,33 +79,57 @@ public class SyncEventCreationManager {
      *                           accepted and false if it needs to be resubmitted later
      */
     public SyncEventCreationManager(
+            @NonNull final PlatformContext platformContext,
+            @NonNull final Time time,
             @NonNull final EventCreator creator,
             @NonNull final EventCreationRule eventCreationRules,
             @NonNull final Function<GossipEvent, Boolean> eventConsumer) {
 
         this.creator = Objects.requireNonNull(creator);
         this.eventCreationRules = Objects.requireNonNull(eventCreationRules);
         this.eventConsumer = Objects.requireNonNull(eventConsumer);
+
+        phase = new PhaseTimerBuilder<>(platformContext, time, "platform", EventCreationStatus.class)
+                .enableFractionalMetrics()
+                .setInitialPhase(PAUSED)
+                .setMetricsNamePrefix("eventCreation")
+                .build();
     }
 
     /**
      * Attempt to create an event. If successful, attempt to pass that event to the event consumer.
      */
     public void maybeCreateEvent() {
-        if (paused || !eventCreationRules.isEventCreationPermitted()) {
+        if (paused) {
+            phase.activatePhase(PAUSED);
+            return;
+        }
+
+        if (!eventCreationRules.isEventCreationPermitted()) {
+            phase.activatePhase(eventCreationRules.getEventCreationStatus());
             return;
         }
 
         tryToSubmitMostRecentEvent();
         if (mostRecentlyCreatedEvent != null) {
             // Don't create a new event until the previous one has been accepted.
+            phase.activatePhase(PIPELINE_INSERTION);
             return;
         }
 
+        phase.activatePhase(ATTEMPTING_CREATION);
+
         mostRecentlyCreatedEvent = creator.maybeCreateEvent();
-        if (mostRecentlyCreatedEvent != null) {
+        if (mostRecentlyCreatedEvent == null) {
+            // The only reason why the event creator may choose not to create an event
+            // is if there are no eligible parents.
+            phase.activatePhase(NO_ELIGIBLE_PARENTS);
+        } else {
             eventCreationRules.eventWasCreated();
             tryToSubmitMostRecentEvent();
```

Comment:
Does the time spent in `tryToSubmitMostRecentEvent()` need to be measured as a discrete value?  This method call (lines 113 and 129) is impacting whatever phase was active when the containing method is called and the `ATTEMPTING_CREATION` metric.

Reply:
I've moved the status `phase.activatePhase(PIPELINE_INSERTION);` into `tryToSubmitMostRecentEvent()` and rearranged ordering so that we correctly enter the state `PIPELINE_INSERTION` when we get stuck inserting something.